fastai/losses.py

```diff
@@ -70,55 +104,95 @@ class FocalLossFlat(BaseLoss):
     https://arxiv.org/pdf/1708.02002.pdf. Note the class weighting factor in the paper, alpha, can be
     implemented through pytorch `weight` argument passed through to F.cross_entropy.
     """
-    y_int = True
+    y_int = True # y interpolation
     @use_kwargs_dict(keep=True, weight=None, reduction='mean')
-    def __init__(self, *args, gamma=2.0, axis=-1, **kwargs):
+    def __init__(self,
+        *args,
+        gamma:float=2.0, # Weighting factor to address class imbalance
+        axis:int=-1, # Axis self.decodes and self.activation occurs on
+        **kwargs
+    ):
```

Comment:
Focal loss' gamma had the weights doc string, not gamma's
```suggestion
    def __init__(self,
        *args,
        gamma:float=2.0, # Focusing parameter. Higher values down-weight easy examples' contribution to loss
        axis:int=-1, # Input class axis
        **kwargs
    ):
```

Reply:
Is `Class axis` appropriate?  I wasn't quite sure what Input referred to so I just did Class axis, but let me know if that changes the intent

Reply:
Input label axis would probably be the most accurate

Reply:
I used your suggestion here, but I don't understand why we use the term `input` in this case.

Reply:
On second look, I misread something and you are right, it shouldn't be input

Reply:
Ok, either `Label axis` or `Class axis` then?

Reply:
I'm leaning towards `Class axis` but I'll leave it up to you